torch_xla/csrc/xla_sharding_util.cpp

```diff
@@ -313,6 +314,10 @@ ShardingUtil::InputHandler(
     std::vector<std::string> devices) {
   tsl::profiler::TraceMe activity("InputHandler",
                                   tsl::profiler::TraceMeLevel::kInfo);
+  // TODO: use vector<DataPtr> directly since DataPtr is already sharded
```

Comment:
Wait, I think that it would be nicer to have all the changes in the runtime wrapper, and PyTorch/XLA still calls the same API, work with the same data prepresentation for SPMD with the same sementic. I.e., have the same-level of abstraction in our wrapper client, so the underlying details can be hidden (DataPtr goes to a single device buffer) or exposed (ArrayPtr goes to block across devices), instead of asking to assume DataPtr is now sharded or not shsarded. We really shoulnd't reimplement PyTorch/XLA for this, IFRT was designed to support that.

Reply:
PjRtShardedData abstraction (or call it IfRtShardedData) stores shards as Array, which used to be a vector<PjRtData>. But the abstraction and its level-of-abstraction stays the same, and our implementation still oeprates the same.

Reply:
In this case, we should change the signature of `ExecuteReplicated` to accept one vector of sharded tensors. Any assembling/disassembling can happen in the computation client.

I'll separate any ComputationClient API changes into a new PR to ensure they work with PJRT.